src/native/crt.c

```diff
@@ -156,7 +168,9 @@ static void s_jvm_table_remove_jvm_for_env(JNIEnv *env) {
 }
 
 JNIEnv *aws_jni_acquire_thread_env(JavaVM *jvm) {
-    aws_rw_lock_rlock(&s_jvm_table_lock);
+    if (aws_rw_lock_try_rlock(&s_jvm_table_lock)) {
+        return NULL;
```

Comment:
trivial: I would add `aws_raise_error(AWS_ERROR_JAVA_CRT_JVM_DESTROYED);`
otherwise the errors is just going to be something like `WAAA_MY_LOCK_FAILED`

also, add comments about why we're using try-lock here, because the reasoning is complicated

Reply:
I left the unimplemented error alone in my update.